ghcide/src/Development/IDE/Core/Rules.hs

```diff
@@ -824,11 +824,11 @@ isHiFileStableRule = defineEarlyCutoff $ RuleNoDiagnostics $ \IsHiFileStable f -
 
 displayTHWarning :: LspT c IO ()
 displayTHWarning
-  | isMac && not hostIsDynamic = do
+  | not isWindows && not hostIsDynamic = do
       LSP.sendNotification SWindowShowMessage $
         ShowMessageParams MtInfo $ T.unwords
-          [ "This HLS binary does not support Template Haskell."
-          , "Follow the [instructions](" <> templateHaskellInstructions <> ")"
+          [ "This HLS binary could not support Template Haskell."
+          , "Follow this [instructions](" <> templateHaskellInstructions <> ")"
```

Comment:
I think the old version read better to me.

Reply:
the does -> could part? or the -> this? the first one is a semantic change cause th could work for some cases

Reply:
Then I'd use "might" there

Reply:
I used might but I read somewhere that might denotes less probability and after @pepeiborra concern I wanted to make stronger the fact it could not work

Reply:
original version restored